lib/generators/scenic/view/view_generator.rb

```diff
@@ -85,10 +85,22 @@ def previous_definition
         Scenic::Definition.new(plural_file_name, previous_version)
       end
 
+      def plural_file_name
+        @plural_file_name ||= file_name.pluralize.gsub(".", "_")
+      end
+
       def destroying?
         behavior == :revoke
       end
 
+      def formatted_plural_name
+        if plural_name.include?(".")
+          "\"#{plural_name}\""
```

Comment:
If we do that then the string interpolation won't work...

```ruby
irb(main):001:0> var = 'Hi!'
=> "Hi!"
irb(main):002:0> '"#{var}"'
=> "\"\#{var}\""
```

If you don't want the escaping we could use delimiters like this: `%("#{plural_name}")` and the interpolation will still work. I don't love that syntax personally, but it's another option.